File: Python/fib.py
import sys
print("A program that allows a user to calculate the Fibonacci Sequence from an interval\n")

while True: #Main prorgam loop
    while True: #Keep asking until the user enters a valid input.
        print("Enter the Nth Fibonacci number you wish to calculate\n(such as 5,50,1000,9999), or QUIT to quit:")
        response = input().upper()
        #print(f"The number you enter is {response}")
    
        if response == 'QUIT':
            print('Thanks for playing!')
            sys.exit()

        if response.isdecimal() and int(response) !=0:
            nth = int(response)
            break
        
        print('Please enter a number greater than 0, or QUIT')
    print()

        #Handle the special cases if the user enter 1 or 2
    if nth ==1:
            print('0\n')
            print('The #1 Fibonacci number is 0.')
            continue

    elif nth==2:
            print('0,1\n')
            print('The #2 Fibonacci number is 1.')
            continue

        #Display a warning if the user entered a large number:
    if nth >=10000:
            print('WARNING: This will take a while to display on the \nscreen. If you want to quit this program before it is\n done, press Ctrl-C.')
            print('screen. If you want to quit this program before it is')
            print('done, press Ctrl-C.')
            input('Press Enter to begin...')
            
        #Calculate the Nth Fibonacci number:
    secondToLastNumber = 0
    lastNumber = 1
    fibNumbersCalculated = 2
    print('0, 1, ',end='')  #Display the first two Fiboncacci numbers.

        #Display all the later numbers of the Fibonacci numbers.
    while True:
            nextNumber = secondToLastNumber + lastNumber
            fibNumbersCalculated +=1

            #Display the next number in the sequence:
            print(nextNumber, end = '')
            #Check if we've found the Nth number the user wants:
            if fibNumbersCalculated == nth: 
                print()
                print()
                print('The #', fibNumbersCalculated, ' Fibonacci ', 'number is ', nextNumber, sep='')
                break 

             # Print a comma in between the sequence numbers:
            print(', ',end='')

            #Shift the last two numbers:
            secondToLastNumber = lastNumber
            lastNumber = nextNumber
    

 

    
    

File: C++/fib.cpp
#include "fib.h"
#include <iostream>
#include <string>

Fib::Fib(){
    sequence = nullptr;
    last =0;
}
Fib::Fib(int n){
  last =  n;
  sequence = new int[n];
}

Fib::~Fib(){
    if(last>0){
        delete sequence;
    }
}

void Fib::calculateSequence(){
    if(last==0){
        return;
    }
    if(last==1){
         sequence[0] = 1;
         return;
    }
    if(last==2){
        sequence[0]=1;
        sequence[1]=1;
        return;
    }
    if(last>=2){
        sequence[0]=1;
        sequence[1]=1;
        int prev =0;
        int pred =1;
        int i =2;
        while(i!=last){
            sequence[i] = sequence[pred] + sequence[prev];
            pred = prev;
            prev = i;
            i++;
        }
    }
   
}
void Fib::setLast(int newLast){
    if(newLast>0)
    {
    last = newLast;
    sequence = new int[newLast];
    }
}

string Fib::displaySequence(){
    string output= "";
    if(last==0){
        return "The sequence is empty because the number of elements you entered to calculate is 0.";
    }
    for(int i =0; i<last-1;i++){
        output+= to_string(sequence[i]) + ", ";
    }
    output += to_string(sequence[last-1]); 
    return output;
}

string Fib::search(int n){
    if(last==0){
        return "The sequence is empty. As such " +to_string(n)+" is not in the sequence.";
        
    }
    if(n<=0 && last>0){
        return to_string(n) + " is not in the list";
    }
    for(int i =0; i<last;i++){
        if(sequence[i]==n){
            return to_string(n) + " is in the sequence.";
        }
    }
    return to_string(n) + " is not in the sequence.";
}
